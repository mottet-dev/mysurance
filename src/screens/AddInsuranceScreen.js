import React, { Component } from 'react';
import { View, StatusBar } from 'react-native';
import { Header } from 'react-native-elements';

import { backgroundColor, lightColor, strongColor } from '../constants';

import AddInsuranceForm from '../components/AddInsuranceForm/AddInsuranceForm';

class AddInsuranceScreen extends Component {
    render() {
        const { screenStyle, headerOuterStyle, componentTitleStyle } = styles;

        return (
            <View style={screenStyle}>
                <StatusBar hidden />
                <Header
                    outerContainerStyles={headerOuterStyle}
                    centerComponent={{ text: 'Add Insurance', style: componentTitleStyle }}
                />
                <AddInsuranceForm />
            </View>
        );
    }
}

const styles = {
    screenStyle: {
        backgroundColor,
        flex: 1,
    },
    headerOuterStyle: {
        backgroundColor: strongColor
    },
    componentTitleStyle: {
        color: lightColor,
        fontSize: 20
    }
}

export default AddInsuranceScreen;