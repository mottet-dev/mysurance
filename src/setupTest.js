/*  
    Configure Enzyme to work with React 16 
    Is called before every test
*/

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export default configure({ adapter: new Adapter() });