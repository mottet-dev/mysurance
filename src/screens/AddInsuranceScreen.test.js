import '../setupTest';
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from 'react-native-elements';

import AddInsuranceScreen from './AddInsuranceScreen';

describe('<AddInsuranceScreen />', () => {
    const screen = shallow(<AddInsuranceScreen />);

    it('renders properly', () => {
        expect(screen).toMatchSnapshot();
    });

    it('contains 1 <AddInsuranceForm />', () => {
        expect(screen.find('View').children('AddInsuranceForm').length)
            .toBe(1);
    });

    it('contains 1 <Header />', () => {
        expect(screen.find('View').children(Header).length)
            .toBe(1);
    });
});