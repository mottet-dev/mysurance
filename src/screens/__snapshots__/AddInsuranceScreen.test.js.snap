// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddInsuranceScreen /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddInsuranceScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          hidden={true}
          showHideTransition="fade"
        />,
        <Header
          centerComponent={
            Object {
              "style": Object {
                "color": "#FBF7F7",
                "fontSize": 20,
              },
              "text": "Add Insurance",
            }
          }
          outerContainerStyles={
            Object {
              "backgroundColor": "#DC4712",
            }
          }
        />,
        <AddInsuranceForm />,
      ],
      "style": Object {
        "backgroundColor": "#F1E9E3",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "hidden": true,
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centerComponent": Object {
            "style": Object {
              "color": "#FBF7F7",
              "fontSize": 20,
            },
            "text": "Add Insurance",
          },
          "outerContainerStyles": Object {
            "backgroundColor": "#DC4712",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            hidden={true}
            showHideTransition="fade"
          />,
          <Header
            centerComponent={
              Object {
                "style": Object {
                  "color": "#FBF7F7",
                  "fontSize": 20,
                },
                "text": "Add Insurance",
              }
            }
            outerContainerStyles={
              Object {
                "backgroundColor": "#DC4712",
              }
            }
          />,
          <AddInsuranceForm />,
        ],
        "style": Object {
          "backgroundColor": "#F1E9E3",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "hidden": true,
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "centerComponent": Object {
              "style": Object {
                "color": "#FBF7F7",
                "fontSize": 20,
              },
              "text": "Add Insurance",
            },
            "outerContainerStyles": Object {
              "backgroundColor": "#DC4712",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
